graphql/util.go

```diff
@@ -620,13 +620,14 @@ func ModifyVersion(version model.Version, user user.DBUser, proj *model.ProjectR
 	case SetPriority:
 		var projId string
 		if proj == nil {
-			projId, err := model.GetIdForProject(version.Identifier)
+			foundProjId, err := model.GetIdForProject(version.Identifier)
 			if err != nil {
 				return http.StatusNotFound, errors.Errorf("error getting project ref: %s", err)
 			}
-			if projId == "" {
+			if foundProjId == "" {
 				return http.StatusNotFound, errors.Errorf("project for %s came back nil: %s", version.Branch, err)
 			}
+			projId = foundProjId
 		} else {
 			projId = proj.Id
 		}
```

Comment:
The issue was that `projId` was technically being redeclared within the `if` statement because of the `:=` syntax. I think using this syntax doesn’t change the value of `projId` outside of the `if` statement, but instead declares a new variable within this scope. 

For this reason, `projId` was always an empty string `""` (the default value) when constructing the `gimlet.PermissionOpts` struct, so it was not possible to correctly determine if a user had the appropriate permissions for setting patch priority on a given project.

Reply:
good find!!! The reason this is happening in the if statement is because we're using := (which creates a new variable) instead of = (which assigns to the variable); you could continue to use projId if you wanted within the if statement and set it this way instead (`projId, err = ...`).

Reply:
Actually we shouldn't have to do this at all; version.Identifier already equals the project ID (which is confusing since it's labeled identifier), so we could just set projId directly to version.Identifier

Reply:
I'll give it a go! Thank you for reviewing my PR right before your vacation 🔥